docs/content/commands/npm-init.md

```diff
@@ -71,6 +72,68 @@ Generate it without having it ask any questions:
 $ npm init -y
 ```
 
+### Workspaces support
+
+It's possible to create a new workspace within your project by using the
+`workspace` config option. When using `npm init -w <dir>` the cli will
+create the folders and boilerplate expected while also adding a reference
+to your project `package.json` `"workspaces": []` property in order to make
+sure that new generated **workspace** is properly set up as such.
+
+Given a project with no workspaces, e.g:
+
+```
+.
++-- package.json
+```
+
+You may generate a new workspace using the legacy init:
+
+```bash
+$ npm init -w packages/a
+```
+
+That will generate a new folder and `package.json` file, while also updating
+your top-level `package.json` to add the reference to this new workspace:
```

Comment:
Why if my intention (as is/will be common) is to make all child workspaces be “packages/*”? Is there any way to let npm init do this for me, or must i manually edit package.json?

Reply:
you have to manually edit the `package.json` to put it there, but once it's there `npm init` always performes a check to avoid adding extra lines to that array, e.g:

```
# Starts out with an empty workspaces property:
$ cat package.json
{
  "workspaces": []
}

# Generates a new workspace, will place the path to the workspace folder there:
$ npm init -w packages/a
$ cat package.json
{
  "workspaces": [
    "packages/a"
  ]
}

# Manually edit package.json to use a more permissive glob pattern:
$ sed s#packages/a#packages/*#g package.json
$ cat package.json
{
  "workspaces": [
    "packages/*"
  ]
}

# Generates a second workspace in which the path to the folder is comprised by the existing glob pattern:
$ npm init -w packages/b

# Project package.json is now not touched/modified:
$ cat package.json
{
  "workspaces": [
    "packages/*"
  ]
}
```

Reply:
That's a great middle ground.

It seems like we might want a way to support this use case without manual editing, no?

Reply:
yeah, let's remain open to improving it in the future 😊 but for now I agree this is a very good middle ground.